packages/editor-ui/src/modules/credentials.ts

```diff
@@ -135,6 +148,12 @@ const module: Module<ICredentialsState, IRootState> = {
 
 			return credentials;
 		},
+		fetchScopes: async (context: ActionContext<ICredentialsState, IRootState>, credentialType: string): Promise<string[]> => {
+			const scopes = await getScopes(context.rootGetters.getRestApiContext, credentialType);
```

Comment:
question... why did not we simply extend credential types with scope? why did we need a separate api request for this?

Reply:
In my reasoning, because there is only a subset of OAuth cred types that have scopes, and of these, only those that are selected by the user in HTTPRN v2 are needed. I think we discussed this with Alex before? Let me know otherwise.

Reply:
not really a fan of this design. might be worth discussing with Ben

Reply:
Moved in this branch from endpoint to store.